### PR TITLE
*/*: non-games.eclass consumers of base.eclass

### PR DESCRIPTION
**Pull request by thesamesam**

### CODE REVIEW

File: dev-java/jspeex/jspeex-0.9.7-r1.ebuild [hunk 1]
```diff
@@ -1,43 +1,47 @@
...
 
-S="${WORKDIR}/${PN}"
+src_prepare() {
+	default
 
-java_prepare() {
-	epatch "${FILESDIR}"/${P}-remove-junit-report.patch
-	epatch "${FILESDIR}"/${P}-remove-proguard-taskdef.patch
+	eapply "${FILESDIR}"/${P}-remove-junit-report.patch
+	eapply "${FILESDIR}"/${P}-remove-proguard-taskdef.patch
```
Comment (SoapGentoo) [inline dev-java/jspeex/jspeex-0.9.7-r1.ebuild:42]: why not PATCHES?

File: eclass/openib.eclass [hunk 2]
```diff
@@ -60,7 +67,14 @@ block_other_ofed_versions() {
 	done
 }
 
-OFED_BASE_VER=$(get_version_component_range 1-3 ${OFED_VER})
+case ${EAPI:-0} in
+	0|1|2|3|4|5|6)
+		OFED_BASE_VER=$(get_version_component_range 1-3 ${OFED_VER})
```
Comment (SoapGentoo) [inline eclass/openib.eclass:72]: eapi7-ver rather?

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: dev-java/jspeex/jspeex-0.9.7-r1.ebuild [hunk 3]
```diff
@@ -1,43 +1,47 @@
...
 CDEPEND="dev-java/ant-core:0"
-
-RDEPEND="${CDEPEND}
-	>=virtual/jre-1.4"
-
-DEPEND="${CDEPEND}
-	>=virtual/jdk-1.4
+RDEPEND="
+	${CDEPEND}
+	>=virtual/jre-1.4
```
Comment (fordfrog) [inline dev-java/jspeex/jspeex-0.9.7-r1.ebuild:22]: should be changed to `>=virtual/jre-1.8:*` and done in a revbump instead. though not related to eapi 7, it's a part of the ebuild cleanup :-) java 1.4 is not supported by java 11, in fact the lowest supported version by java 11 is 1.6 but we put there the lowest version supported on gentoo which is 1.8.

File: dev-java/jspeex/jspeex-0.9.7-r1.ebuild [hunk 4]
```diff
@@ -1,43 +1,47 @@
...
+"
+DEPEND="
+	${CDEPEND}
 	dev-java/junit:0
+	>=virtual/jdk-1.4
```
Comment (fordfrog) [inline dev-java/jspeex/jspeex-0.9.7-r1.ebuild:27]: the same as in `RDEPEND`, should be changed to `virtual/jdk-1.8:*`.

File: dev-java/jspeex/jspeex-0.9.7-r1.ebuild [hunk 5]
```diff
@@ -1,43 +1,47 @@
...
+	default
 
-java_prepare() {
-	epatch "${FILESDIR}"/${P}-remove-junit-report.patch
-	epatch "${FILESDIR}"/${P}-remove-proguard-taskdef.patch
+	eapply "${FILESDIR}"/${P}-remove-junit-report.patch
+	eapply "${FILESDIR}"/${P}-remove-proguard-taskdef.patch
 
-	find . -name "*.jar" -delete || die "Failed to remove bundled libraries."
+	find . -name "*.jar" -delete || die "Failed to remove bundled libraries"
```
Comment (fordfrog) [inline dev-java/jspeex/jspeex-0.9.7-r1.ebuild:44]: should use [java-pkg_clean](https://github.com/gentoo/gentoo/blob/master/eclass/java-utils-2.eclass#L2924) instead.

Comment (thesamesam) [reply dev-java/jspeex/jspeex-0.9.7-r1.ebuild:44]: Thank you!

File: dev-java/jspeex/jspeex-0.9.7-r1.ebuild (hunk 4 above)
Comment (thesamesam) [reply dev-java/jspeex/jspeex-0.9.7-r1.ebuild:27]: Thanks!

File: dev-java/jspeex/jspeex-0.9.7-r1.ebuild (hunk 3 above)
Comment (thesamesam) [reply dev-java/jspeex/jspeex-0.9.7-r1.ebuild:22]: Excellent point, thank you @fordfrog for your comments!

File: dev-java/jspeex/jspeex-0.9.7-r1.ebuild (hunk 1 above)
Comment (thesamesam) [reply dev-java/jspeex/jspeex-0.9.7-r1.ebuild:42]: Oh, duh. Thanks!

**Comment by SoapGentoo** [comment]
Obsolete with 5b884a467f475b3581cdcdc92d1ddcf4ac4d26ad

**Pull request closed without merging**